### DISCUSSION TITLE
Add basepath support to Hydrate CLI

### DISCUSSION THREAD

**Discussion by clintandrewhall** [architect/architect#1250] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

I'm using Typescript, so I compile my routes to `/src`.  So `arc hydrate --update` isn't looking in the right place to check for dependencies.

**Describe the solution you'd like**

Consider including `--basepath` to the Hydrate CLI

**Describe alternatives you've considered**

I have a script at the moment that will check my `lib` directory and run `npm install` in each one.  It appears `hydrate` has better logic.

**Additional context or notes**
https://github.com/clintandrewhall/itemize/blob/6615232f2eb2adb5df1bcf5c8ba234500900ceb5/scripts/route-install.js

**Comment by filmaj** [comment]
I wonder if this could be fixed with something like script 'hooks'. Arc already has this functionality via the sandbox init scripts (thank you @ryanblock). Maybe we should expose more arc lifecycle events using this same method for things like hydration?
